cf-oscore/src/test/java/org/eclipse/californium/oscore/ContextRederivationTest.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2018 RISE SICS and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Rikard Höglund (RISE SICS)
+ *    
+ ******************************************************************************/
+package org.eclipse.californium.oscore;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.security.SecureRandom;
+
+import org.eclipse.californium.core.CoapClient;
+import org.eclipse.californium.core.CoapResponse;
+import org.eclipse.californium.core.Utils;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.elements.exception.ConnectorException;
+import org.junit.Test;
+import org.eclipse.californium.core.coap.CoAP.Code;
+import org.eclipse.californium.core.coap.CoAP.ResponseCode;
+import org.eclipse.californium.cose.AlgorithmID;
+
+/**
+ * Class that implements test of functionality for re-derivation of contexts.
+ * As detailed in Appendix B.2. of the OSCORE draft:
+ * https://tools.ietf.org/html/draft-ietf-core-object-security-16#appendix-B.2
+ *
+ * This can for instance be used when one device has lost power and information
+ * about the mutable parts of a context (e.g. sequence number) but retains information
+ * about static parts (e.g. master secret)
+ * 
+ * To be ran together with the HelloWorldServer
+ */
+public class ContextRederivationTest {
+
```

Comment:
```
@ClassRule
public static CoapNetworkRule network = new CoapNetworkRule(CoapNetworkRule.Mode.DIRECT, CoapNetworkRule.Mode.NATIVE);
```

For unit test with Socket communication, a `CoapNetworkRule` is required.

Reply:
Done in commit cdecae7c41deeb2a8f26ee011fe6747a97f4acf0